io/io/src/TFile.cxx

```diff
@@ -1334,7 +1334,7 @@ std::pair<TList *, Int_t> TFile::GetStreamerInfoListImpl(bool lookupSICache)
       char *buffer = new char[fNbytesInfo+1];
       char *buf    = buffer;
       Seek(fSeekInfo);
-      if (ReadBuffer(buf,fNbytesInfo)) {
+      if (ReadBuffer(buf,fNbytesInfo+1)) {
```

Comment:
Why do we need the +1 ?

Reply:
because the buffer is fNbytesInfo+1 characters long?

Reply:
The data content is definitively only fNbytesInfo and the +1 read past the 'end'.   I am not sure why the (very first version of this) code allocated one extra byte (see f896460257a).

Reply:
So in order to avoid potentially different hashes for the same record because of this extra byte, the right length to use is fNbytesInfo, right?